catalog/dags/elasticsearch_cluster/create_proportional_by_source_staging_index/create_proportional_by_source_staging_index_dag.py

```diff
@@ -0,0 +1,188 @@
+"""
+# Create Proportional By Source Staging Index DAG
+
+This DAG is used to create a new staging Elasticsearch index that is a subset
+of a staging source index, such that the proportions of records by source in
+the new index is equal to the proportions of records by source in the source
+index.
+
+
+Required Dagrun Configuration parameters:
+
+* media_type:         The media type for which to create a new index.
+* percentage_of_prod: A float indicating the proportion of items to take from each
+                      source from the total amount existing in the staging
+                      source index
+
+Optional params:
+
+* source_index: An existing staging Elasticsearch index to use as the basis for
+                the new index. If not provided, the index aliased to
+                `<media_type>-filtered` will be used.
+
+## When this DAG runs
+
+This DAG is on a `None` schedule and is run manually.
+
+## Race conditions
+
+Because this DAG runs on the staging ingestion server and staging elasticsearch
+cluster, it does _not_ interfere with the `data_refresh` or
+`create_filtered_index` DAGs.
+
+However, as the DAG operates on the staging API database it will exit
+immediately if any of the following DAGs are running:
+* `staging_database_restore`
+* `recreate_full_staging_index`
+* `create_new_staging_es_index`
+"""
+from datetime import datetime, timedelta
+
+from airflow.decorators import dag
+from airflow.models import Variable
+from airflow.models.param import Param
+
+from common import elasticsearch as es
+from common import slack
+from common.constants import (
+    AUDIO,
+    DAG_DEFAULT_ARGS,
+    MEDIA_TYPES,
+    STAGING,
+)
+from common.sensors.utils import prevent_concurrency_with_dags
+from database.staging_database_restore.constants import (
+    DAG_ID as STAGING_DB_RESTORE_DAG_ID,
+)
+from elasticsearch_cluster.create_new_es_index.create_new_es_index_types import (
+    CREATE_NEW_INDEX_CONFIGS,
+)
+from elasticsearch_cluster.create_proportional_by_source_staging_index import (
+    create_proportional_by_source_staging_index as create_index,
+)
+from elasticsearch_cluster.recreate_staging_index.recreate_full_staging_index import (
+    DAG_ID as RECREATE_STAGING_INDEX_DAG_ID,
+)
+
+
+DAG_ID = "create_proportional_by_source_staging_index"
+
+
+@dag(
+    dag_id=DAG_ID,
+    default_args=DAG_DEFAULT_ARGS,
+    schedule=None,
+    start_date=datetime(2024, 1, 31),
+    tags=["database", "elasticsearch"],
+    max_active_runs=1,
+    catchup=False,
+    doc_md=__doc__,
+    params={
+        "media_type": Param(
+            default=AUDIO,
+            enum=MEDIA_TYPES,
+            description="The media type for which to create the index.",
+        ),
+        "percentage_of_prod": Param(
+            default=0.5,
+            type="number",
+            minimum=0,
+            maximum=1,
```

Comment:
I tried 0 for the percentage out of curiosity and it failed in the `trigger_reindex` task with the following error:
```
elasticsearch.BadRequestError: BadRequestError(400, 'x_content_parse_exception', '[1:181] [reindex] failed to parse field [max_docs]')
```

It doesn't make sense to try creating an index of zero items anyway, so an exclusive minimum would be more appropriate here (I learned [this validation](https://json-schema.org/understanding-json-schema/reference/numeric#range) exists recently!).

```suggestion
        "percentage_of_prod": Param(
            default=0.5,
            type="number",
            exclusiveMinimum=0,
            maximum=1,
```

Reply:
Cool, TIL!